src/simulation/deviceInterface/stepperMotor/stepperMotor.cpp

```diff
@@ -0,0 +1,259 @@
+ @return bool
+ @param t [s] Time the method is called
+*/
+bool StepperMotor::isInStepFirstHalf(double t) { return (t < this->ts && std::abs(this->ts - t) > 1e-5); }
+
+/*! This method computes the motor states during the first half of each step.
+ @return void
+ @param t [s] Time the method is called
+*/
+void StepperMotor::computeStepFirstHalf(double t) {
+    if (this->intermediateThetaRef > this->intermediateThetaInit) {
+        this->thetaDDot = this->thetaDDotMax;
+    } else {
+        this->thetaDDot = -this->thetaDDotMax;
+    }
+    this->thetaDot = this->thetaDDot * (t - this->tInit);
+    this->theta = this->a * (t - this->tInit) * (t - this->tInit) + this->intermediateThetaInit;
+}
+
+/*! This method determines if the motor is in the second half of a step.
+ @return bool
+ @param t [s] Time the method is called
+*/
+bool StepperMotor::isInStepSecondHalf(double t) {
+    return ((t >= this->ts || std::abs(this->ts - t) < 1e-5) && std::abs(this->tf - t) > 1e-5);
+}
+
+/*! This method computes the motor states during the second half of each step.
+ @return void
+ @param t [s] Time the method is called
+*/
+void StepperMotor::computeStepSecondHalf(double t) {
+    if (this->intermediateThetaRef > this->intermediateThetaInit) {
+        this->thetaDDot = -this->thetaDDotMax;
+    } else {
+        this->thetaDDot = this->thetaDDotMax;
+    }
+    this->thetaDot = this->thetaDDot * (t - this->tInit) - this->thetaDDot * (this->tf - this->tInit);
+    this->theta = this->b * (t - this->tf) * (t - this->tf) + this->intermediateThetaRef;
+}
+
+/*! This method computes the motor states when a step is complete.
+ @return void
+ @param t [s] Time the method is called
+*/
+void StepperMotor::computeStepComplete(double t) {
+    this->stepComplete = true;
+    this->thetaDot = 0.0;
+    this->theta = this->intermediateThetaRef;
+    this->tInit = t;
+
+    // Update the motor step count
+    if (this->intermediateThetaRef > this->intermediateThetaInit) {
+        this->stepCount++;
+    } else {
+        this->stepCount--;
+    }
+
+    // Update the actuationComplete boolean variable only when motor actuation is complete
+    if ((this->stepCount == this->stepsCommanded) && !this->interruptMsg) {
+        this->actuationComplete = true;
+    }
+}
+
+/*! Getter method for the initial motor angle.
+ @return double
+*/
+double StepperMotor::getThetaInit() const { return this->thetaInit; }
+
+/*! Getter method for the motor step angle.
+ @return double
+*/
+double StepperMotor::getStepAngle() const { return this->stepAngle; }
+
+/*! Getter method for the motor step time.
+ @return double
+*/
+double StepperMotor::getStepTime() const { return this->stepTime; }
+
+/*! Getter method for the maximum motor angular acceleration.
+ @return double
+*/
+double StepperMotor::getThetaDDotMax() const { return this->thetaDDotMax; }
+
+/*! Setter method for the initial motor angle.
+ @return void
```

Comment:
These `@return void` statements raise warnings in sphinx and should be removed.  I pushed a [squash] labeled commit that does this.  CI Documentation now completes where before it failed.